docs/writing-rules/pattern-syntax.md

```diff
@@ -473,16 +493,43 @@ Many programmers don't really see the difference between `foo()` and `foo();`. T
     Note that in some programming languages such as Python, which does not use semicolons as a separator or terminator, the difference between expressions and statements is even more confusing. Indentation in Python matters, and a newline after `foo()` is really the same than `foo();` in other programming languages such as C.
 
 
-## Partial statements
+## Partial expressions
 
-Partial statements are not valid patterns. For example, the following are invalid:
+Partial expressions are not valid patterns. For example, the following is invalid:
 
 ```text
 pattern: 1+
 ```
 
-```text
-pattern: if $CONDITION:
+A complete expression is needed (like `1 + $X`)
+
+## Partial statements
+
+Partial statements are partially supported. For example,
+you can just match the header of a conditional with `if ($E)`,
```

Comment:
I am not fully aware about where this is particularly useful. When is this preferred to `if ($E) ...` ?

Reply:
Is this only because you want to match that fragment of the code and nothing else?

Reply:
Can be useful when used in a pattern-inside, to restrict the context in which to search code

Reply:
I can probaly add that as a comment